packages/patternfly-4/react-table/src/components/Table/BodyWrapper.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { css } from '@patternfly/react-styles';
+import styles from '@patternfly/react-styles/css/components/Table/table';
+import { IRow, IRowData, IExtraData, IHeaderRow } from './Table';
+import { mapOpenedRows } from './utils/headerUtils';
+
+export interface BodyWrapperProps {
+  children?: React.ReactNode;
+  mappedRows?: IRow[];
+  rows?: IRow[];
+  onCollapse?: (event: React.MouseEvent, rowIndex: number, isOpen: boolean, rowData: IRowData, extraData: IExtraData) => undefined;
+  tbodyRef?: React.Ref<any>;
```

Comment:
Good point... it looks like TypeScript does not like passing a `Function` to a `tbody`'s ref prop:
```
Type 'Function | ((instance: any) => void) | RefObject<any>' is not assignable to type 'LegacyRef<HTMLTableSectionElement>'.
  Type 'Function' is not assignable to type 'LegacyRef<HTMLTableSectionElement>'.
    Type 'Function' is not assignable to type 'string'.ts(2322)
index.d.ts(97, 9): The expected type comes from property 'ref' which is declared here on type 'DetailedHTMLProps<HTMLAttributes<HTMLTableSectionElement>, HTMLTableSectionElement>'
```

But I do think we should still accept a `Function` for the legacy use case... so for now, adding:

```
tbodyRef?: React.Ref<any> | Function;
```
and casting it beneath:
```
ref={tbodyRef as React.Ref<any>}
```

Let me know if you think of a better option. Not all that familiar w/ this myself!

Reply:
That sounds good to me. How does that cast behave if the user passes a function? is that still an error? I'm not too familiar with type assertions yet.

Reply:
Well - i believe it works the same way as passing `ref` prop now in JS. It can be a function or a ref type at runtime. This is just casting the type for static type analysis. Please feel free to double check me on this, however I am seeing that passing a function to `RowWrapper` from our existing `editableRowWrapper` works similarly...and does not give warnings or errors at runtime in Gatsby...

https://github.com/patternfly/patternfly-react/blob/master/packages/patternfly-4/react-inline-edit-extension/src/components/InlineEdit/editableRowWrapper.js#L21